a1/README.md

```diff
@@ -3268,6 +3269,98 @@ Use [Gulp](http://gulpjs.com) or [Grunt](http://gruntjs.com) for creating automa
+
+  *Why?*: Because these are features that have been made available in the latest versions of JavaScript, but did not exist natively when the Angular helper functions were first written.
+
+  Replace `angular.forEach()` and `angular.isArray()`:
+
+  ```javascript
+  /* avoid */
+  var arr = [1, 2, 3];
+  var obj = { hello: 'Hello', world: 'World' };
+
+  angular.forEach(arr, function(value) {
+    console.log(value);
+  });
+
+  angular.forEach(obj, function(value) {
+    console.log(value);
+  });
+
+  angular.isArray(arr); // true
+  angular.isArray(obj); // false
+  ```
+
+  ```javascript
+  /* recommended */
+  var arr = [1, 2, 3];
+  var obj = { hello: 'Hello', world: 'World' };
+
+  // iterate through arrays
+  arr.forEach(function(value) {
+    console.log(value);
+  });
+
+  // iterate through objects
+  Object.keys(obj)
+    .forEach(function(key) {
+      console.log(obj[key]);
+    });
+
+  // alternative lodash helper to iterate through objects
+  _.forEach(obj, function(value) {
+    console.log(value);
+  });
+
+  // native js functionality
+  Array.isArray(arr); // true
+  Array.isArray(obj); // false
+  ```
+
+  Favor usage of JavaScript features (in this case available starting with ES2015) over `angular.copy()` and `angular.extend()`.
+
+  ```javascript
+  /* avoid */
+  var arr1 = [1, 2, 3];
+  var arr2 = angular.copy(arr1); // [1, 2, 3]
+
+  var obj1 = { hello: 'Hello' };
+  var obj2 = { world: 'World' };
+  var obj3 = angular.extend({}, obj1, obj2); // {hello: 'Hello', world: 'World'}
+  ```
+
+  ```javascript
+  /* recommended */
+  var arr1 = [1, 2, 3];
+  var arr2 = [...arr1]; // [1, 2, 3]
+
+  var obj1 = { hello: 'Hello' };
+  var obj2 = { world: 'World' };
+  var obj3 = Object.assign({}, obj1, obj2); // {hello: 'Hello', world: 'World'}
+  ```
+
+  *Why?*: These are commonly known features of JavaScript and makes code more intuitive to read.
+
+  *Why?*: This will ease the process of upgrading to Angular 2 as it does not ship with these helper functions.
+
+
+  - Note: favor usage of functions that are available in JavaScript natively, then resort to using a utility library like [lodash](https://lodash.com/docs) when what you need is not available.
+
+  - Note: you might need to use a [polyfill](https://babeljs.io/docs/usage/polyfill/) to take advantage of the latest features of JavaScript for better browser support.
+
+  - Note: Be sure to understand the differences in implementation when using a native JavaScript feature over an Angular helper. Some of Angular's helpers have some nuances that are relevant to its internal API (ie. dealing with `$$hashKey`, etc).
```

Comment:
The example with the (undocumented, private) `$$hashKey` will sound scary to the uninitiated :smile:
I would rather give the example of Angular helpers being more "forgiving" (e.g. not throwing on `undefined`/`null` values).